File: responses/AEEQEN-J2J/2_sum_of_certain_multiples.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(X):
    # write your code in Python
    if X > 0:
        X = str(X)[::-1]
    else:
        #X = str(X)[0:1] + str(X)[2:-1:-1]
        X = str(X)[0] + str(X)[-1:0:-1]
#    print(X)
    return(int(X))

File: responses/YJ7PGK-HKS/2_sum_of_certain_multiples.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def sum_first_n(n)
  n * (n + 1) / 2
end

def solution(x)
  three_count = (x - 1) / 3
  five_count = (x - 1) / 5
  fifteen_count = (x - 1) / 15
  
  three = 3 * sum_first_n(three_count)
  five = 5 * sum_first_n(five_count)
  fifteen = 15 * sum_first_n(fifteen_count)
  
  three + five - 2 * fifteen
end
